homeassistant/components/qnap/sensor.py

```diff
@@ -52,14 +43,78 @@
 ATTR_UPTIME = "Uptime"
 ATTR_VOLUME_SIZE = "Volume Size"
 
-CONF_DRIVES = "drives"
-CONF_NICS = "nics"
-CONF_VOLUMES = "volumes"
 
-NOTIFICATION_ID = "qnap_notification"
-NOTIFICATION_TITLE = "QNAP Sensor Setup"
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up entry."""
+    coordinator = QnapCoordinator(hass, config_entry)
+    await coordinator.async_refresh()
+    if not coordinator.last_update_success:
+        raise PlatformNotReady
+    uid = config_entry.unique_id
+    sensors: list[QNAPSensor] = []
+
+    sensors.extend(
+        [
+            QNAPSystemSensor(coordinator, description, uid)
+            for description in _SYS_SENSORS
+        ]
+    )
+
+    sensors.extend(
+        [QNAPCPUSensor(coordinator, description, uid) for description in _CPU_SENSORS]
+    )
+
+    sensors.extend(
+        [
+            QNAPMemorySensor(coordinator, description, uid)
+            for description in _MEM_SENSORS
+        ]
+    )
+
+    # Network sensors
+    sensors.extend(
+        [
+            QNAPNetworkSensor(coordinator, description, uid, nic)
+            for nic in coordinator.data["system_stats"]["nics"]
+            for description in _NET_SENSORS
+        ]
+    )
+
+    # Drive sensors
+    sensors.extend(
+        [
+            QNAPDriveSensor(coordinator, description, uid, drive)
+            for drive in coordinator.data["smart_drive_health"]
+            for description in _DRI_SENSORS
+        ]
+    )
 
-_SYSTEM_MON_COND: tuple[SensorEntityDescription, ...] = (
+    # Volume sensors
+    sensors.extend(
+        [
+            QNAPVolumeSensor(coordinator, description, uid, volume)
+            for volume in coordinator.data["volumes"]
+            for description in _VOL_SENSORS
+        ]
+    )
+    async_add_entities(sensors)
+
+
+def round_nicely(number):
+    """Round a number based on its size (so it looks nice)."""
+    if number < 10:
+        return round(number, 2)
+    if number < 100:
+        return round(number, 1)
+
+    return round(number)
+
+
+_SYS_SENSORS: tuple[SensorEntityDescription, ...] = (
```

Comment:
As I've said before, please remove all changes to the `SensorEntityDescription`
They do not belong in this PR.

Please also make sure that you keep them above `async_setup_entry` to make review easier.

Reply:
Not yet addressed